### PR TITLE
feat(cli): add `lb4 openapi-spec` command to save openapi spec to json or yaml

### PR DESCRIPTION
**Pull request by raymondfeng**
## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [x] API Documentation in code was updated
- [x] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [x] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

**Comment by bajtos** [comment]
Have you considered to put this code into a new file (and a new `npm run` script) inside the LB4 application project? See how we are implementing `npm run migrate`:

- [template for `src/migrate.ts`](https://github.com/strongloop/loopback-next/blob/master/packages/cli/generators/app/templates/src/migrate.ts.ejs)
- npm script: https://github.com/strongloop/loopback-next/blob/1013e4f2790e925c8e5d086ef248f420dd1f8585/packages/cli/generators/project/templates/package.json.ejs#L53
- PR that introduced this feature: https://github.com/strongloop/loopback-next/pull/2094

File: packages/cli/generators/openapi-spec/index.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    if (this.options.out) {
+      this.outFile = this.options.out;
+      return;
+    }
+    const prompts = [
+      {
+        type: 'input',
+        name: 'outFile',
+        message: g.f('File name for the OpenAPI spec:'),
+        default: 'dist/openapi.json',
```
Comment (dhmlau) [inline packages/cli/generators/openapi-spec/index.js:48]: is it possible to make the default at the root of the project?

**Comment by raymondfeng** [comment]
@bajtos I was debating between the two approaches. Here is another PR - https://github.com/strongloop/loopback-next/pull/5521.

In the future, maybe we can generate `src/workspace.ts` to boot the application and inspect artifacts such as models, datasources, etc.

**Comment by bajtos** [comment]
Thank you @raymondfeng for opening #5521, I like it more than the CLI-based proposal.

**Comment by raymondfeng** [comment]
Favor https://github.com/strongloop/loopback-next/pull/5521

**Pull request closed without merging**